pkg/compute/executor_buffer.go

```diff
@@ -48,8 +50,8 @@ type ExecutorBuffer struct {
 	delegateService            Executor
 	callback                   Callback
 	running                    map[string]*bufferTask
-	enqueued                   map[string]*bufferTask
-	enqueuedList               []string
+	queuedTasks                *collections.PriorityQueue[*bufferTask]
+	enqueuedIDs                map[string]struct{}
```

Comment:
does it make sense to create a `HashedPriorityQueue` or `PriorityHashedQueue` under `collections` package that wraps a queue and a map for us and implements a queue with a `contains()` method that returns in `O(1)`? Might just simplify our testing if we extract this logic from the executor buffer